aten/src/THC/generic/THCTensorKthValue.cu

```diff
@@ -0,0 +1,122 @@
+      getTensorInfo<int64_t, THCudaLongTensor, INDEX_T>(state, indices);  \
+                                                                          \
+    /* We use these structures solely to find the offset to */            \
+    /* each slice we are operating on */                                  \
+    inputInfo.sizes[dim] = 1;                                             \
+    kthValueInfo.sizes[dim] = 1;                                          \
+    indicesInfo.sizes[dim] = 1;                                           \
+                                                                          \
+    /* Collapse all other dims */                                         \
+    int collapseInputDim = inputInfo.collapseDims(dim);                   \
+    int collapseKthValueDim = kthValueInfo.collapseDims(dim);             \
+    int collapseIndicesDim = indicesInfo.collapseDims(dim);               \
+                                                                          \
+    int64_t inputSlices = 1;                                              \
+    for (int i = 0; i < inputInfo.dims; ++i) {                            \
+      inputSlices *= inputInfo.sizes[i];                                  \
+    }                                                                     \
+    int64_t kthValueSlices = 1;                                           \
+    for (int i = 0; i < kthValueInfo.dims; ++i) {                         \
+      kthValueSlices *= kthValueInfo.sizes[i];                            \
+    }                                                                     \
+                                                                          \
+    dim3 grid;                                                            \
+    if (!THC_getGridFromTiles(inputSlices, grid)) {                       \
+      THError("Slice to select is too large");                            \
+    }                                                                     \
+                                                                          \
+    dim3 block(std::min(THCRoundUp(sliceSize, (int64_t) 32), (int64_t) 1024)); \
+                                                                          \
+    /* This is used as a template parameter to calculate indices. */      \
+    /* We only specialize it if all collapsed dim sizes are the */        \
+    /* same; otherwise, we use -1 which is the specialization */          \
+    /* parameter for arbitrary dimensions */                              \
+    int allDims = inputInfo.dims;                                         \
+    if (kthValueInfo.dims != allDims || indicesInfo.dims != allDims) {    \
+      allDims = -1;                                                       \
+    }                                                                     \
+                                                                          \
+    RUN_DIM(INDEX_T);
```

Comment:
nit: unindent one level to match the code above and not the macro indentation

Reply:
done